src/main/java/io/cdap/plugin/gcp/common/GCPUtils.java

```diff
@@ -49,11 +64,55 @@ public static ServiceAccountCredentials loadServiceAccountCredentials(String pat
     }
   }
 
+  public static ServiceAccountCredentials loadServiceAccountCredentials(String content,
+                                                                        boolean isServiceAccountFilePath)
+    throws IOException {
+    if (isServiceAccountFilePath) {
+      return loadServiceAccountCredentials(content);
+    }
+    InputStream jsonInputStream = new ByteArrayInputStream(content.getBytes());
+    return ServiceAccountCredentials.fromStream(jsonInputStream);
+  }
+
+  public static String extractPrivateKey(ServiceAccountCredentials credentials) {
+    return String.format(PRIVATE_KEY_WRAP,
+                         Base64.getEncoder().encodeToString(credentials.getPrivateKey().getEncoded()));
+  }
+
+  public static Map<String, String> generateAuthProperties(String serviceAccount,
+                                                           boolean isServiceAccountFilePath,
+                                                           String... keyPrefix) throws IOException {
```

Comment:
Do we have a case where we need to support multiple key prefixes?

Reply:
Yes, here is a case: https://github.com/data-integrations/google-cloud/blob/48a86cecc1cdf7e45b95bfbe76b19662754b8f2a/src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtil.java#L155-L158